hutch_python/startup_script.py

```diff
@@ -0,0 +1,23 @@
+"""
+Initialize the ELogPoster callback and subscribe it to the RunEngine
+
+Startup script files to be run after ipython is initialized via:
+``c.InteractiveTerminalApp.exec_files``
+
+Code here will take executed after both IPython has been loaded and
+the namespace has been populated with hutch objects.
+
+These will be run as standalone python files, and should not be
+imported from.
+"""
+
+import IPython
+from nabs.callbacks import ELogPoster
+
+from hutch_python.utils import safe_load
+
+if __name__ == '__main__':
+    with safe_load('ELogPoster'):
```

Comment:
(It's OK as-is, just a suggestion)

Seeing non-trivial blocks inside of `if __name__ == "__main__"` always makes me a bit uneasy, though.
To that end, I think this could still go in a function, then it could be made more explicit what's to be part of the global namespace:

```python
def _configure_elog_poster():
    with safe_load(...):
        elog = globals().get("elog", None)
        elogc = ...
    return elogc

if __name__ == "__main__":
   elogc = _configure_elog_poster()
```

Reply:
I also meant to include the imports inside the function here so they don't get brought into the session as well (unless they're already there somehow).

Reply:
I like this, lets me actually lint those lines.